src/packagers/JSPackager.js

```diff
@@ -93,6 +93,17 @@ class JSPackager extends Packager {
     return name;
   }
 
+  dedupeKey(asset) {
+    // cannot rely *only* on generated JS for deduplication because paths like
+    // `../` can cause 2 identical JS files to behave differently depending on
+    // where they are located on the filesystem
+    let deps = Array.from(asset.dependencies.keys()).map(dep =>
```

Comment:
Instead of using `asset.dependencies` here, you could use `asset.depAssets` which is a list of resolved assets rather than dependencies. So you won't even need to call `resolveFilename`, it will already have the resolved filename as its `name` property.

Reply:
@devongovett I think it still needs `resolveFilename` as the `name` property contains only the relative file name.
But something like this should work:
```js
    let deps = Array.from(asset.depAssets.keys()).map(dep => {
      return this.bundler.resolver.resolveFilename(dep.name, asset.name)
    });
```

Reply:
@henri-hulski It should probably be `...resolveFilename(dep, asset.name)` in your suggestion as `dep` is a string.

Reply:
@felixrabe here `dep` is not a string. I've logged it out and also tested this fix. It works fine.

Reply:
Ah right, depAssets is a Map, and its key is not a string but a dep (see Bundler.js where it does `asset.depAssets.set(...)`). Still need to get used to Maps having non-string keys it seems :)